index.js

```diff
@@ -29,6 +29,21 @@ module.exports = (rootDir, config = "faucet.config.js", // eslint-disable-next-l
 				watchDirs.join(separator));
 
 		watcher = makeWatcher(watchDirs);
+		watcher.on("error", err => {
+			if(err.code === "ERR_TOO_MANY_FILES") {
+				console.error([
+					"You are watching too many files:",
+					"Please use the watchDirs configuration.",
+					"",
+					/* eslint-disable max-len */
+					"https://github.com/faucet-pipeline/faucet-pipeline#configuration-for-file-watching"
+					/* eslint-enable max-len */
+				].join("\n"));
```

Comment:
How about this:

```javascript
console.error("There are too many files being monitored, " +
		"please use the `watchDirs` configuration setting:\n" +
		// eslint-disable-next-line max-len
		"https://github.com/faucet-pipeline/faucet-pipeline#configuration-for-file-watching");
process.exit(1);
```

(Also note the backticks around "watchDirs".)

More generally, I don't like that we're four levels deep here, so we might consider moving the watcher initialization into a separate function.